plugins/logs/plugin.go

```diff
@@ -385,6 +415,37 @@ func (p *Plugin) bufferChunk(buffer *logBuffer, bs []byte) {
 	}
 }
 
+func (p *Plugin) maskEvent(ctx context.Context, event *EventV1) error {
+
+	query := ast.NewBody(ast.NewExpr(ast.NewTerm(p.config.maskDecisionRef)))
+
+	r := rego.New(
+		rego.ParsedQuery(query),
+		rego.Input(event),
+		rego.Compiler(p.manager.GetCompiler()),
+		rego.Store(p.manager.Store),
+		rego.Runtime(p.manager.Info),
+	)
+
+	rs, err := r.Eval(ctx)
+	if err != nil {
+		return err
+	} else if len(rs) == 0 {
+		return nil
+	}
+
+	ptrs, err := resultValueToPtrs(rs[0].Expressions[0].Value)
```

Comment:
Is it safe to assume `Expressions[0]` will exist?

Reply:
Yes, it's safe because the query contains exactly one expression and this is how the value of the expression is accessed. Alternatively we could construct an expression like `data.system.log.mask = x` and then lookup the x binding but the same logic would apply.

Reply:
Ahh ok, that makes sense. I missed that `parsePathToRef` wouldn't allow someone to configure an empty query or something.